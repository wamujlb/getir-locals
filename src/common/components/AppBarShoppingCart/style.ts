import { Theme } from '@mui/material';
import { createStyles, makeStyles } from '@mui/styles';

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      borderRadius: 0,
      height: 48,
      [theme.breakpoints.up('sm')]: {
        height: 76,
        padding: theme.spacing(2, 3),
      },
    },
  })
);
