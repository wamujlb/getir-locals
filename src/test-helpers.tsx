import React from 'react';
import { ThemeProvider } from '@mui/styles';

import { theme } from 'theme/theme';

export const WrappedWithTheme = ({ children }: any) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};
