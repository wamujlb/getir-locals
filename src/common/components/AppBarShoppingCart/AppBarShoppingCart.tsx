import React from 'react';
import { Button, ButtonProps } from '@mui/material';

import { usePrice } from 'common/hooks/usePrice';
import ShoppingCartIcon from '../icons/ShoppingCart/ShoppingCart';
import { useStyles } from './style';

export type AppBarShoppingCartProps = {
  value: number;
} & ButtonProps;

const AppBarShoppingCart = ({ value }: AppBarShoppingCartProps) => {
  const classes = useStyles();
  const price = usePrice({ value });

  return (
    <Button
      disableElevation
      color="secondary"
      variant="contained"
      classes={classes}
      startIcon={<ShoppingCartIcon />}
    >
      {price}
    </Button>
  );
};

AppBarShoppingCart.displayName = 'AppBarShoppingCart';

export default AppBarShoppingCart;
