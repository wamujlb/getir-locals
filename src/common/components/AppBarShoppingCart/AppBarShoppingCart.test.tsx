import React from 'react';
import { cleanup, render } from '@testing-library/react';

import AppBarShoppingCart from './AppBarShoppingCart';
import { WrappedByTheme } from 'test-helpers';

describe('#AppBarShoppingCart', () => {
  afterEach(() => {
    cleanup();
  });

  it('should match snapshot when price is 0', () => {
    const screen = render(
      <WrappedByTheme>
        <AppBarShoppingCart value={0} />
      </WrappedByTheme>
    );

    expect(screen.container).toMatchSnapshot();
  });

  it('should match snapshot when price is not 0', () => {
    const screen = render(
      <WrappedByTheme>
        <AppBarShoppingCart value={123112} />
      </WrappedByTheme>
    );

    expect(screen.container).toMatchSnapshot();
  });
});
