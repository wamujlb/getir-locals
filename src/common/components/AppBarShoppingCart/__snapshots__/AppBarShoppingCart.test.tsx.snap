// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#AppBarShoppingCart should match snapshot when price is 0 1`] = `
<div>
  <button
    class="makeStyles-root-1 MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-s42839-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3.9,7.8h16.2V22H3.9V7.8z M9.1,1.9h5.8L16,3.1v4.7l-1.2,0V3.1H9.2v4.7L8,7.8V3.1L9.1,1.9z"
        />
      </svg>
    </span>
    ₺ 0.00
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`#AppBarShoppingCart should match snapshot when price is not 0 1`] = `
<div>
  <button
    class="makeStyles-root-2 MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-s42839-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3.9,7.8h16.2V22H3.9V7.8z M9.1,1.9h5.8L16,3.1v4.7l-1.2,0V3.1H9.2v4.7L8,7.8V3.1L9.1,1.9z"
        />
      </svg>
    </span>
    ₺ 123,112.00
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
